cmd/cli/certificate_rotate.go

```diff
@@ -0,0 +1,352 @@
+)
+
+const rotateDesc = `
+This command rotates the OSM Root Certificate
+`
+
+const meshRotateExample = `
+# Rotate the mesh root certificate that is Active in the osm-system namespace
+osm alpha certificate rotate -d -y
+`
+
+type rotateCmd struct {
+	out io.Writer
+
+	meshName        string
+	newTrustDomain  string
+	clientSet       kubernetes.Interface
+	configClient    osmConfigClient.Interface
+	prompt          bool
+	deleteOld       bool
+	waitForRotation time.Duration
+	mrcFilePath     string
+	mrcName         string
+	prompter        func(a ...any) (int, error)
+}
+
+func newCertificateRotateCmd(out io.Writer) *cobra.Command {
+	rotate := &rotateCmd{
+		out: out,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "rotate",
+		Short:   "rotate the MeshRootCertificate",
+		Long:    rotateDesc,
+		Example: meshRotateExample,
+		RunE: func(_ *cobra.Command, _ []string) error {
+			config, err := settings.RESTClientGetter().ToRESTConfig()
+			if err != nil {
+				return fmt.Errorf("error fetching kubeconfig: %w", err)
+			}
+
+			configClient, err := configClientset.NewForConfig(config)
+			if err != nil {
+				return fmt.Errorf("could not access Kubernetes cluster, check kubeconfig: %w", err)
+			}
+			rotate.configClient = configClient
+
+			clientset, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				return fmt.Errorf("could not access Kubernetes cluster, check kubeconfig: %w", err)
+			}
+			rotate.clientSet = clientset
+			rotate.prompter = fmt.Scanln
+
+			return rotate.run()
+		},
+	}
+
+	f := cmd.Flags()
+
+	f.StringVar(&rotate.meshName, "mesh-name", defaultMeshName, "Name of the mesh to upgrade")
+	f.StringVarP(&rotate.newTrustDomain, "trust-domain", "t", "", "If specified the new cert will use this trust domain. Works with certificate provider tresor")
+	f.BoolVarP(&rotate.prompt, "accept", "y", false, "when specified it will not prompt for user input")
+	f.BoolVarP(&rotate.deleteOld, "delete", "d", false, "when specified it will delete the old MeshRootCertificate, otherwise the MeshRootCertificate will remain as inactive")
+	f.DurationVarP(&rotate.waitForRotation, "wait", "w", 60*time.Second, "Time to wait for certificate to propagate to all components at each step")
+	f.StringVarP(&rotate.mrcFilePath, "file", "f", "", "File to use for the new MRC. When not supplied the settings from the existing MRC are copied")
+	f.StringVarP(&rotate.mrcName, "meshrootcertificate", "c", "", "Existing MeshRootCertificate to rotate to active role using the name in currently configured namespace")
+	return cmd
+}
+
+func (r *rotateCmd) promptForContinue(msg string) bool {
+	if !r.prompt {
+		var rotate string
+		fmt.Fprintf(r.out, msg+" y/n:")
+		_, err := r.prompter(rotate)
+		if err != nil {
+			fmt.Printf("warning: error reading value from user: %s\n", err)
+			return true
+		}
+
+		if rotate != "y" {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (r *rotateCmd) run() error {
+	// TODO check if MRC feature is enabled (isMRCEnabled)
```

Comment:
https://github.com/openservicemesh/osm/pull/5204

Reply:
added the issue number here, as well as the next line